src/docs/output-targets/custom-elements.md

```diff
@@ -75,19 +74,81 @@ export const config: Config = {
 | `auto-define-custom-elements` | A component and its children will be automatically defined with the `CustomElementRegistry` when the component's module is imported.                                                                                                                                                                                                                                                                      |
 | `single-export-module`        | All component and custom element definition helper functions will be exported from the `index.js` file in the output directory (see [Defining Exported Custom Elements](#defining-exported-custom-elements) for more information on this file's purpose). This file can be used as the root module when distributing your component library, see [below](#distributing-custom-elements) for more details. |
 
-> Note: At this time, components created not using JSX may not be automatically
-> defined. This is a known limitation of the API and users should be aware of
-> it
+> Note: At this time, components created not using JSX may not be automatically defined. This is a known limitation of the API and users should be aware of it.
+
+### dir
+
+_default: `'dist/components'`_
+
+This config option allows you to change the output directory where the compiled output for this output target will be written.
+
+### empty
+
+_default: `true`_
+
+If `true`, this config option will remove the contents of the [output directory](#dir) between builds.
+
+### externalRuntime
+
+_default: `true`_
+
+When `true`, this flag results in the following behaviors:
+
+1. Minification will follow what is specified in the [Stencil config](docs/config#minifyJs).
+2. Filenames will not be hashed
+3. All imports from packages under `@stencil/core/*` will be marked as external and therefore not included in the generated Rollup bundle.
 
 ### generateTypeDeclarations
 
+_default: `true`_
+
 By default, Stencil will generate type declaration files (`.d.ts` files) as a part of the `dist-custom-elements` output target through the `generateTypeDeclarations` field on the target options. Type declaration files will be placed in the `dist/types` directory in the root of a Stencil project. At this time, this output destination is not able to be configured.
 
 Setting this flag to `false` will not generate type declaration files for the `dist-custom-elements` output target.
 
-This flag defaults to `true` when omitted from a Stencil configuration file.
+> When set to generate type declarations, Stencil respects the export behavior selected via `customElementsExportBehavior` and generates type declarations specific to the content of the generated `component.d.ts` file.
```

Comment:
is it specific to the generated `component.d.ts` or rather that the types that it will generate for `component.d.ts` are appropriate for what is exported given the `customElementsExportBehavior` ? Maybe I'm not understanding something correctly here.

Reply:
Yeah I did make this confusing and botched it a bit. Pushed a change to hopefully make that clearer